sncosmo/builtins.py

```diff
@@ -811,6 +811,38 @@ def load_snemo(relpath, name=None, version=None):
                              version=ver, meta=meta)
 
 
+#SUGAR models
+def load_sugarmodel(relpath, name=None, version=None):
+    rep_name = os.path.expanduser('~/.astropy/cache/sncosmo/models/')
+    try:
+        listdir = os.listdir(rep_name)
+    except:
+        os.system('mkdir -p ~/.astropy/cache/sncosmo/models/')
+        listdir = os.listdir(rep_name)
+    if 'sugar' not in listdir:
+        import requests
+        connect204 = requests.get('http://supernovae.in2p3.fr/sugar_template/sugar.tar.gz')
+        file_out_name = os.path.join(rep_name, 'sugar.tar.gz')
+        open(file_out_name, 'wb').write(connect204.content)
+        connect204.close()
+        os.system('tar -zxvf %s -C %s'%((file_out_name, rep_name)))
+        os.system('rm %s'%(file_out_name))
+
+    abspath = os.path.join(rep_name, 'sugar')
+    return SUGARSource(modeldir=abspath, name=name, version=version)
```

Comment:
Instead of this implementation of `load_sugarmodel` you can do something very similar to `load_salt2model` in this file. (This requires adding a line to the `redirects.json` file in sncosmo.github.io, similar to [this one](https://github.com/sncosmo/sncosmo.github.io/blob/master/data/redirects.json#L10).)

This has several advantages:
- Takes care of untaring and unzipping
- Show download progress
- Generally more robust
- If the files ever move from the `supernova.in2p3.fr` server, we can update the `redirects.json` file and fix the problem immediately, instead of releasing a new version of sncosmo and requiring users to update.
- We don't use requests as a dependency
- Don't use `rm` or `tar` as these won't generally work on Windows.

Reply:
It was done by Florian.